sync/src/com/kirinpatel/Launcher.java

```diff
@@ -0,0 +1,205 @@
+package com.kirinpatel;
+
+import com.google.common.collect.ImmutableList;
+import com.kirinpatel.gui.PlaybackPanel;
+import com.kirinpatel.net.Client;
+import com.kirinpatel.net.Server;
+import com.kirinpatel.net.User;
+import com.kirinpatel.util.UIMessage;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.kirinpatel.gui.PlaybackPanel.PANEL_TYPE.CLIENT;
+import static com.kirinpatel.gui.PlaybackPanel.PANEL_TYPE.SERVER;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.StandardOpenOption.APPEND;
+import static java.nio.file.StandardOpenOption.CREATE;
+
+public final class Launcher extends JFrame {
+
+    private static Launcher INSTANCE;
+    private static AtomicBoolean isInstanceSet = new AtomicBoolean(false);
+
+    static void setInstance() {

```

Comment:
Don't need to do set and get instance I think. Just make instance a public static final field and construct it on declaration. (public static final INSTANCE = new Launcher()).